packages/twenty-front/src/modules/object-record/record-table/components/RecordTableEmptyState.tsx

```diff
@@ -10,24 +12,51 @@ import {
   AnimatedPlaceholderEmptyTextContainer,
   AnimatedPlaceholderEmptyTitle,
 } from '@/ui/layout/animated-placeholder/components/EmptyPlaceholderStyled';
+import { useCombinedViewFilters } from '@/views/hooks/useCombinedViewFilters';
+import { useRecoilValue } from 'recoil';
 
 type RecordTableEmptyStateProps = {
   objectNameSingular: string;
   objectLabel: string;
   createRecord: () => void;
   isRemote: boolean;
+  isSoftDeleteActive: boolean;
+  recordTableId: string;
 };
 
 export const RecordTableEmptyState = ({
   objectNameSingular,
   objectLabel,
   createRecord,
   isRemote,
+  isSoftDeleteActive,
+  recordTableId,
 }: RecordTableEmptyStateProps) => {
   const navigate = useNavigate();
   const { totalCount } = useFindManyRecords({ objectNameSingular, limit: 1 });
   const noExistingRecords = totalCount === 0;
 
+  const { removeCombinedViewFilter } = useCombinedViewFilters(recordTableId);
+  const { tableFiltersState } = useRecordTableStates(recordTableId);
+
+  const tableFilters = useRecoilValue(tableFiltersState);
+
+  const [, toggleSoftDeleteFilterState] = useHandleToggleTrashColumnFilter({
+    objectNameSingular,
+    viewBarId: recordTableId,
+  });
+
+  const handleRemoveSoftDeleteFilter = async () => {
+    removeCombinedViewFilter(
+      tableFilters.find(
+        (filter) =>
+          filter.definition.label === 'Deleted at' &&
+          filter.operand === 'isNotEmpty',
+      )?.id ?? '',
+    );
+    toggleSoftDeleteFilterState(false);
+  };
+
   const [title, subTitle, Icon, onClick, buttonTitle] = isRemote
```

Comment:
Please refactor this part by creating one variable for each variable, so it's more readable in a sequential order.